src/extractor/extraction_containers.cpp

```diff
@@ -380,25 +405,15 @@ void ExtractionContainers::PrepareEdges(ScriptingEnvironment &scripting_environm
             ++edge_iterator;
             continue;
         }
-        if (edge_iterator->result.osm_target_id > node_iterator->node_id)
-        {
-            ++node_iterator;
-            continue;
-        }
 
-        BOOST_ASSERT(edge_iterator->result.osm_target_id == node_iterator->node_id);
-        BOOST_ASSERT(edge_iterator->weight_data.speed >= 0);
-        BOOST_ASSERT(edge_iterator->source_coordinate.lat !=
-                     util::FixedLatitude{std::numeric_limits<std::int32_t>::min()});
-        BOOST_ASSERT(edge_iterator->source_coordinate.lon !=
-                     util::FixedLongitude{std::numeric_limits<std::int32_t>::min()});
+        const auto node_coord = util::Coordinate(all_nodes_list[id_iter->second].lon,
+                                                 all_nodes_list[id_iter->second].lat);
```

Comment:
Also this - previously, coordinates were cache-local in the node iterator, no additional lookup was required.
Because the edge IDs aren't hilbert order sorted, we're essentially doing random access to the `all_nodes_list`.